libavfilter/dnn_filter_common.c

```diff
@@ -17,6 +17,37 @@
  */
 
 #include "dnn_filter_common.h"
+#include "libavutil/avstring.h"
+
+static char **separate_expression(DnnContext *ctx, const char *expr, const char *val_sep, int *separated_nb)
+{
+    char *val, **parsed_vals = NULL;
+    int val_num = 0;
+    if (!expr || !val_sep) {
+        av_log(ctx, AV_LOG_ERROR, "could not parse string for model\n");
+        return NULL;
+    }
+
+    if (!parsed_vals)
+        parsed_vals = av_mallocz(sizeof(*parsed_vals));
+
+    do {
+        val = av_get_token(&expr, val_sep);
+        if(strlen(val)) {
+            parsed_vals[val_num] = val;
+            val_num++;
+            parsed_vals = av_realloc_f(parsed_vals, val_num + 1, sizeof(*parsed_vals));
+        }
+        if (*expr) {
+            expr++;
+        }
+    } while(*expr);
+
+    parsed_vals[val_num] = NULL;
+    *separated_nb = val_num;
```

Comment:
add a null check for separated_nb

Reply:
i thought the separated_nb is a address of an int object(&), does it need a null check as well?

Reply:
separated_nb is a function parameter, it is possible that function caller passes NULL

Reply:
if separated_nb must be valid, we can add a check at the beginning

Reply:
all three of expr, val_sep, separated_nb must be valid. what do you mean by saying 'add a check at the begining'?
add a single 'if(!separated_nb)' OR change current code into 'if (!separated_nb || !expr || !val_sep)'?